src/VisualStudio/Core/Impl/ProjectSystem/CPS/TempPECompiler.cs

```diff
@@ -65,7 +71,7 @@ public async Task<bool> CompileAsync(IWorkspaceProjectContext context, string ou
                     // Turn off any warnings as errors just in case
                     .WithGeneralDiagnosticOption(ReportDiagnostic.Suppress)
                     .WithReportSuppressedDiagnostics(false)
-                    .WithSpecificDiagnosticOptions(null)
+                    .WithSpecificDiagnosticOptions(ImmutableDictionary<string, ReportDiagnostic>.Empty)
```

Comment:
@dotnet/roslyn-compiler: WithSpecificDiagnosticOptions was annotated as accepting non-null, but this code implies that was allowed. Was the intent to discourage it?

Reply:
This code change has been moved to #41292.